src/components/reservation/ReservationModalBody.tsx

```diff
@@ -74,32 +65,27 @@ const ReservationModalBody = ({
   const branches = config<AgencyBranch[]>("branchesConfig", {
     transformer: "jsonParse"
   });
-
-  const mainManifestationType = getManifestationType(mainManifestation);
+  const mainManifestationType = getManifestationType(selectedManifestations[0]);
   const { reservableManifestations } = UseReservableManifestations({
-    manifestations:
-      // TODO: We should investigate why we need to check for parallelManifestation length
-      // because it doesn't seem to reflect the possible types.
-      parallelManifestations && parallelManifestations.length > 0
-        ? parallelManifestations
-        : [mainManifestation],
+    manifestations: selectedManifestations,
     type: mainManifestationType
   });
   const queryClient = useQueryClient();
   const [reservationResponse, setReservationResponse] =
     useState<ReservationResponseV2 | null>(null);
   const [selectedBranch, setSelectedBranch] = useState<string | null>(null);
   const [selectedInterest, setSelectedInterest] = useState<number | null>(null);
-  const faustId = convertPostIdToFaustId(pid);
+  const allPids = getAllPids(selectedManifestations);
+  const faustIds = allPids.map((pid) => convertPostIdToFaustId(pid));
   const { mutate } = useAddReservationsV2();
   const userResponse = useGetPatronInformationByPatronIdV2();
   const holdingsResponse = useGetHoldingsV3({
-    recordid: [faustId]
+    recordid: faustIds
   });
   const { track } = useStatistics();
   const { otherManifestationPreferred } = useAlternativeAvailableManifestation(
     work,
-    mainManifestation.pid
+    selectedManifestations[0].pid
```

Comment:
Please pass the pids for all the provided materials into the hook. We only prefer an alternate material if there is an available one which is different from all ids provided.